### PR TITLE
Small improvements for better use

### PR DESCRIPTION
**Pull request by Xraydylan**
Changes:
- Made `classes` an optional argument with the default class being "1"

Added:
- `disable_legend` argument to optionally not display classes in legend
- `clear_positions()` function to remove all points from a class (or all classes) with external call

### CODE REVIEW

**Comment by Xraydylan** [comment]
Note:
I haven't made changes to the doc so far

File: mpl_point_clicker/_clicker.py [hunk 1]
```diff
@@ -66,35 +74,51 @@ class in *classes*
             colors = [None] * len(self._classes)
         if markers is None:
             markers = ["o"] * len(self._classes)
+        if disable_legend is None:
+            disable_legend = False
+        if disable_legend and len(self._classes) != 1:
+            disable_legend = False
```
Comment (ianhi) [inline mpl_point_clicker/_clicker.py:80]:
```suggestion
```

I understand the urge here - but we shouldn't override the user input here. They may be changing the current class programmatically or with their own gui elements. Also with `disable_legend` default to False we don't need to check for `None`

File: mpl_point_clicker/_clicker.py [hunk 2]
```diff
@@ -47,6 +51,10 @@ class in *classes*
             Line2D objects (from ax.plot) are used to generate the markers.
             line_kwargs will be passed through to all of the `ax.plot` calls.
         """
+        if classes is None:
+            classes = 1
+            disable_legend = True
```
Comment (ianhi) [inline mpl_point_clicker/_clicker.py:56]: Move this below where we turn `classes` into a list so we can check the length and thus have a more robust check

```suggestion
        if len(classes) == 1) and disable_legend is None:
            disable_legend = True
```

**Review by ianhi — CHANGES_REQUESTED** [review]
Hi @Xraydylan thanks for this PR! This is going to be a great change, to my surprise it seems that most people use this library with only one class.

I've requested some coding changes below but looking good to start

**Comment by ianhi** [comment]
the docs build is not your fault. I have an outdated config file that readthedocs finally stopped supporting. I'll fix that in a separate PR

**Comment by ianhi** [comment]
Thank you for the contribution @Xraydylan ! You are actually the first contributor outside of me for this repo :) I really appreciate you taking the time to do this work

**Pull request merged**